File: server/src/models/Anomaly.js
/**
 * Pseudocode:
 * 1) Document Anomaly model shape.
 * 2) Store validation issues for analytics.
 */

/**
 * @typedef {Object} Anomaly
 * @property {string} anomalyId
 * @property {string} eventId
 * @property {"missing_field"|"impossible_order"|"out_of_range"|"unknown"} anomalyType
 * @property {string} detectedBy
 * @property {"open"|"ignored"|"fixed"} resolutionStatus
 * @property {Object<string, unknown>=} details
 */

export const ANOMALY_MODEL_DOCS = {};

File: server/src/models/Anomaly.java
package models;

import java.util.Map;

/**
 * Pseudocode:
 * 1) Document Anomaly model shape.
 * 2) Store validation issues for analytics.
 */
public class Anomaly {
  public String anomalyId;
  public String eventId;
  public String anomalyType;
  public String detectedBy;
  public String resolutionStatus;
  public Map<String, Object> details;
}
